src/components/MyAccount/RequestsTab/FreezeButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react"
+import type { Hold, Patron } from "../../../types/accountTypes"
+import { Button } from "@nypl/design-system-react-components"
+
+const FreezeButton = ({ hold, patron }: { hold: Hold; patron: Patron }) => {
+  const [frozen, setFrozen] = useState(hold.frozen)
+  const [isDisabled, setIsDisabled] = useState(false)
+
+  const handleFreezeClick = async () => {
+    // Disabling button while request happens.
+    setIsDisabled(true)
+    if (frozen) {
```

Comment:
can this if/else be eliminated by using the value of `frozen` for the property `freeze`? the `setFrozen` call too can use `!frozen` in that case.